File: frontend/features/estoque/types/lote.ts
export interface Lote {
  id: number;
  codigo: string;
  data_fabricacao?: string;
  data_validade?: string;
  observacoes?: string;
  created_at?: string;
  updated_at?: string;
}

export interface CreateLoteDTO {
  codigo: string;
  data_fabricacao?: string;
  data_validade?: string;
  observacoes?: string;
}

export interface UpdateLoteDTO {
  codigo?: string;
  data_fabricacao?: string;
  data_validade?: string;
  observacoes?: string;
}

File: backend/app/modules/estoque/models/lote.py
from sqlalchemy import Column, Integer, String, Date, DateTime, Text
from sqlalchemy.sql import func
from app.db.session import Base


class Lote(Base):
    __tablename__ = "lotes"

    id = Column(Integer, primary_key=True, index=True)
    codigo = Column(String(50), nullable=False, unique=True, index=True)
    data_fabricacao = Column(Date, nullable=True)
    data_validade = Column(Date, nullable=True)
    observacoes = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    def __repr__(self):
        return f"<Lote(id={self.id}, codigo='{self.codigo}', validade='{self.data_validade}')>"
